tests/SprykerEcoTest/Zed/Adyen/Business/Facade/AdyenFacadeExecutePostSaveHookTest.php

```diff
@@ -0,0 +1,180 @@
+    }
+
+    /**
+     * @return void
+     */
+    public function testExecutePostSaveHookShouldContainErrorsWhenResponsIsNotSuccess(): void
+    {
+        //Arrange
+        $adyenApiResponseTransfer = (new AdyenApiResponseBuilder([
+            AdyenApiResponseTransfer::IS_SUCCESS => false,
+        ]))
+            ->withMakePaymentResponse([AdyenApiMakePaymentResponseTransfer::RESULT_CODE => static::ADYEN_PAYMENT_STATUS_AUTHORISED])
+            ->withError()
+            ->build();
+
+        $adyenToAdyenApiFacadeBridgeMock = $this->createMock(AdyenToAdyenApiFacadeBridge::class);
+        $adyenToAdyenApiFacadeBridgeMock->method('performMakePaymentApiCall')
+            ->willReturn($adyenApiResponseTransfer);
+
+        $facade = $this->createFacade($adyenToAdyenApiFacadeBridgeMock);
+
+        $orderTransfer = $this->setUpCommandTest(
+            static::PROCESS_NAME_ADYEN_CREDIT_CARD,
+            static::OMS_STATUS_NEW,
+        );
+
+        $spyPaymentAdyen = $this->getSpyPaymentAdyen($orderTransfer);
+        $checkoutResponseTransfer = $this->tester->createCheckoutResponseTransfer();
+
+        //Act
+        $facade->executePostSaveHook(
+            $this->tester->buildQuoteTransfer($spyPaymentAdyen->getReference()),
+            $checkoutResponseTransfer,
+        );
+
+        //Assert
+        $this->assertSame(1, $checkoutResponseTransfer->getErrors()->count());
+    }
+
+    /**
+     * @dataProvider refusalStatusDataProvider
+     *
+     * @param string $refusalStatus
+     *
+     * @return void
+     */
+    public function testExecutePostSaveHookShouldNotContainErrorsWhenResponsIsSuccessAndHasRefusalStatus(string $refusalStatus): void
+    {
+        //Arrange
+        $adyenApiResponseTransfer = (new AdyenApiResponseBuilder([
+            AdyenApiResponseTransfer::IS_SUCCESS => false,
+        ]))
+            ->withMakePaymentResponse([AdyenApiMakePaymentResponseTransfer::RESULT_CODE => static::ADYEN_PAYMENT_STATUS_AUTHORISED])
+            ->withError()
+            ->build();
+
+        $adyenToAdyenApiFacadeBridgeMock = $this->createMock(AdyenToAdyenApiFacadeBridge::class);
+        $adyenToAdyenApiFacadeBridgeMock->method('performMakePaymentApiCall')
+            ->willReturn($adyenApiResponseTransfer);
+
+        $facade = $this->createFacade($adyenToAdyenApiFacadeBridgeMock);
+
+        $orderTransfer = $this->setUpCommandTest(
+            static::PROCESS_NAME_ADYEN_CREDIT_CARD,
+            static::OMS_STATUS_NEW,
+        );
+
+        $spyPaymentAdyen = $this->getSpyPaymentAdyen($orderTransfer);
+        $checkoutResponseTransfer = $this->tester->createCheckoutResponseTransfer();
+
+        //Act
+        $facade->executePostSaveHook(
+            $this->tester->buildQuoteTransfer($spyPaymentAdyen->getReference()),
+            $checkoutResponseTransfer,
+        );
+
+        //Assert
+        $this->assertSame(1, $checkoutResponseTransfer->getErrors()->count());
```

Comment:
Please, add into all test cases extra assert to validate value for `$checkoutResponseTransfer->getIsSuccess()` 
and check CheckoutErrorTransfer properties if it's `false`.